### PR TITLE
deploy a reusable workflow to build the docs

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
This PR has been opened in order to fix issue #15.

### CODE REVIEW

File: .github/workflows/deploy-sphinx-docs-on-ghpages.yml [hunk 1]
```diff
@@ -37,42 +37,12 @@ jobs:
   build:
     
     runs-on: ubuntu-latest
```
Comment (AleCandido) [inline .github/workflows/deploy-sphinx-docs-on-ghpages.yml:39]: This one I believe you can drop, you're only interested in where the action called is running, not the caller :)

**Review by AleCandido — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-sphinx-docs-on-ghpages.yml (hunk 1 above)
Comment (Edoardo-Pedicillo) [reply .github/workflows/deploy-sphinx-docs-on-ghpages.yml]: I have to drop it. Thank you

File: .github/workflows/deploy-ghpages-latest-stable.yml [hunk 2]
```diff
@@ -57,40 +57,14 @@ jobs:
...
   build:
     
-    runs-on: ubuntu-latest
     needs: check-label
-    steps:
+    uses: ./.github/workflows/build-docs.yml
+    with:
+      python-version: ${{ inputs.python-version }}
+      package-manager: ${{ inputs.package-manager }}
+      dependency-path: ${{ inputs.dependency-path }}
```
Comment (AleCandido) [inline .github/workflows/deploy-ghpages-latest-stable.yml:65]: You forgot to pass down `path-doc`: you might get one from the user, so you want to pass it also to the further workflow.

**Review by AleCandido — CHANGES_REQUESTED** [review]

**Comment by Edoardo-Pedicillo** [comment]
@AleCandido, If you agree, we can merge this PR. After this, I can modify the name of the branches in the files `deploy_doc.yml` of Qibo & co and merge them.

**Review by AleCandido — APPROVED** [review]

**Pull request merged**